### PR TITLE
#181339522-API-documentation-using-swagger

### PR DESCRIPTION
**Pull request by Code-250**
### what does this Pull Request do?
this Pull Request sets up the documentation of the barefoot nomad project using swagger.

### description of tasks to be completed
 - setting up the documentation of the project using Swagger
 - documenting the home route using swagger to test the working of the swagger.

### how should this Pull Request be tested?

 - checkout `ch-setup-swagger-documentation-#181339522` branch
 - run in your terminal `yarn run dev` to start the project
 - run in your browser `localhost:port/api-docs` to go to the swagger doc link.

### How should this be tested in Production?

- [Barefoot Namad-documentation-setup](https://staging-bare-ch-setup-s-hkwhna.herokuapp.com/api-docs/)
- 
### pivotal tracker
 - [pivotal tracker link](https://www.pivotaltracker.com/n/projects/2555280)

### screeshots
Screen Shot 2022-02-28 at 16 08 38

- deployed screenshot
Screen Shot 2022-03-03 at 16 15 26

### Questions
N/A

### CODE REVIEW

**Review by Gabin-ishimwe — APPROVED** [review]
good work richard, i liked how you documented each route and imported it in the main document(index.js). it is very neat and clear

**Review by Tsa2341 — APPROVED** [review]
Nice work Richard.

File: .env.example [hunk 1]
```diff
@@ -0,0 +1,2 @@
+PORT=********
+BASE_URL=localhost:port
```
Comment (ncutixavier) [inline .env.example]: - Remove one env.example

File: .env.sample [hunk 2]
```diff
@@ -1,2 +1,2 @@
-DEV_DATABASE_URL=postgres://<db_user>:<db_password>@127.0.0.1:5432/<dev_database_name>
+DEV_DATABASE_URL=postgres://postgres:Richard@Munyemana@127.0.0.1:5432/cabal
```
Comment (ncutixavier) [inline .env.sample]: Remove your credential and replace it with the previous one

File: src/documentation/welcome.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import responses from "./response";
+
+const welcome = {
+  "/home": {
+    get: {
+      tags: ["Welcome"],
+      summary: "Welcome",
+      responses,
```
Comment (ncutixavier) [inline src/documentation/welcome.js:6]: Add only required response not all of them

File: src/documentation/index.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    : process.env.BASE_URL?.split("http://")[1];
+
+const paths = {
+  ...welcome,
+};
+
+const config = {
+  swagger: "2.0",
+  info: {
+    description: "Employee Travel Management",
```
Comment (ncutixavier) [inline src/documentation/index.js]: Change this description

**Review by ncutixavier — CHANGES_REQUESTED** [review]

**Review by ncutixavier — APPROVED** [review]

**Pull request merged**